src/main/java/org/tugraz/sysds/runtime/lineage/LineageItem.java

```diff
@@ -0,0 +1,93 @@
+package org.tugraz.sysds.runtime.lineage;
+
+import java.util.ArrayList;
+
+import org.tugraz.sysds.runtime.instructions.cp.CPOperand;
+
+public class LineageItem {
+    private static int _id_counter = 1;
+
+    private final int _id;
+    private final String _opcode;
+    private final CPOperand _variable;
+    private final ArrayList<LineageItem> _ancestors;
+    private final ArrayList<LineageItem> _children;
+    private boolean _visited = false;
+
+    public LineageItem(CPOperand variable) {
+        _id = LineageItem.getUniqueId();
+        _variable = new CPOperand(variable);
+        _opcode = "";
+        _ancestors = new ArrayList<>();
+        _children = new ArrayList<>();
+    }
+
+    public LineageItem(CPOperand variable, ArrayList<LineageItem> ancestors, String opcode) {
```

Comment:
To make it slightly more general, maybe use `List<LineageItem>` here.

Reply:
I suppose only for the member, because List<> is an interface.

Reply:
well no, define the argument as a List<> which means you can pass a LinkedList or an ArrayList etc from outside.

Reply:
Okay, now I've got it.